### ISSUE TITLE
Using datasets.Metric with Trainer()

### ISSUE THREAD

**Issue by YipingNUS** [huggingface/datasets#1592]
## Using datasets.Metric with Trainer()

Hi team, I was quite surprised in the [Metric documentation](https://huggingface.co/docs/datasets/using_metrics.html) I don't see how it can be used with `Trainer()`. That would be the most intuitive use case instead of having to iterate the batches and add predictions and references to the metric, then compute the metric manually. Ideally, any pre-built metrics can be added to `compute_metrics` argument of `Trainer()` and they will be calculated at an interval specified by `TrainingArguments.evaluation_strategy`. 

Is this option available but just not mentioned in the documentation or it's not possible at the moment? I notice in the [Transformer | Training and fine-tuning](https://huggingface.co/transformers/training.html) tutorial, you are using custom scripts to calculate the accuracy, P/R/F, which are already in the pre-built metrics.

**Comment by lhoestq** [comment]
We are indeed working on the integration with `Trainer` :)
